utils/datasets/cifar10_dataset.py

```diff
@@ -0,0 +1,312 @@
+# Copyright 2019, Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Library for loading and preprocessing CIFAR-10 training and testing data."""
+
+import collections
+from typing import Tuple
+import numpy as np
+import tensorflow as tf
+import tensorflow_federated as tff
+from tensorflow.keras import datasets, layers, models
+
+CIFAR_SHAPE = (32, 32, 3)
+TOTAL_FEATURE_SIZE = 32 * 32 * 3
+NUM_EXAMPLES_PER_CLIENT = 5000
+
+def load_cifar10_federated(
+    num_clients: int = 10, 
+    num_classes: int = 10, 
+    alpha: float = 1, 
+    train_client_batch_size: int = 20, 
+    test_client_batch_size: int = 100):
+    '''
+    Loads the train dataset into a non iid distribution over clients using the
+    sampling method based on LDA, taken from this paper:
+    Measuring the Effects of Non-Identical Data Distribution for
+    Federated Visual Classification (https://arxiv.org/pdf/1909.06335.pdf).
+    '''
+    (train_images, train_labels), (test_images, test_labels) = datasets.cifar10.load_data()
+    client_train_dataset = collections.OrderedDict()
+    client_test_dataset = collections.OrderedDict()
+
+    # for each class in the dataset, there are 10 for CIFAR-10
+    idx_batch = [[] for _ in range(num_clients)]
+
+    for k in range(num_classes):
+        '''
+        This will form the final dataset dictionary for each client
+        For each class(label) sample in a proportion for each client. 
+        The proportion is determined by a dirichlet distribution.
+        '''
```

Comment:
Use '#' for multi-line comments. https://google.github.io/styleguide/pyguide.html#385-block-and-inline-comments

Reply:
Revised. Thanks!